crates/chain-state/src/notifications.rs

```diff
@@ -152,54 +152,34 @@ pub trait ForkChoiceSubscriptions: Send + Sync {
     fn subscribe_finalized_block(&self) -> ForkChoiceNotifications;
 
     /// Convenience method to get a stream of the new safe blocks of the chain.
-    fn safe_block_stream(&self) -> ForkChoiceStream {
-        ForkChoiceStream { st: WatchStream::new(self.subscribe_safe_block().0) }
+    fn safe_block_stream(&self) -> ForkChoiceStream<SealedHeader> {
+        ForkChoiceStream::<SealedHeader> { st: WatchStream::new(self.subscribe_safe_block().0) }
     }
 
     /// Convenience method to get a stream of the new finalized blocks of the chain.
-    fn finalized_block_stream(&self) -> ForkChoiceStream {
-        ForkChoiceStream { st: WatchStream::new(self.subscribe_finalized_block().0) }
-    }
-}
-
-/// A stream of the fork choices in the form of [`SealedHeader`].
-#[derive(Debug)]
-#[pin_project::pin_project]
-pub struct ForkChoiceStream {
-    #[pin]
-    st: WatchStream<Option<SealedHeader>>,
-}
-
-impl Stream for ForkChoiceStream {
-    type Item = SealedHeader;
-
-    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        loop {
-            match ready!(self.as_mut().project().st.poll_next(cx)) {
-                Some(Some(notification)) => return Poll::Ready(Some(notification)),
-                Some(None) => continue,
-                None => return Poll::Ready(None),
-            }
+    fn finalized_block_stream(&self) -> ForkChoiceStream<SealedHeader> {
+        ForkChoiceStream::<SealedHeader> {
+            st: WatchStream::new(self.subscribe_finalized_block().0),
         }
     }
 }
 
-/// A stream for block state watch channels (pending, safe or finalized watchers)
+/// A stream for fork choice watch channels (pending, safe or finalized watchers)
 #[derive(Debug)]
 #[pin_project::pin_project]
-pub struct BlockStateNotificationStream<T> {
+pub struct ForkChoiceStream<T> {
```

Comment:
I actually quite like the name ForkChoiceStream, let me know if this is appropriate for the type or needs renaming